src/core/qfieldcloud/qfieldcloudproject.cpp

```diff
@@ -984,6 +983,30 @@ void QFieldCloudProject::download()
   } );
 }
 
+void QFieldCloudProject::prepareDownloadTransfer( const QString &projectId, const QString &fileName, int fileSize, const QString &cloudEtag )
+{
+  const QString fileKey = QStringLiteral( "%1/%2" ).arg( projectId, fileName );
+  const QString projectDir = QStringLiteral( "%1/%2/%3" ).arg( QFieldCloudUtils::localCloudDirectory(), mUsername, projectId );
+
+  FileTransfer transfer( fileName, fileSize, projectId, cloudEtag );
+  transfer.partialFilePath = QDir( projectDir ).filePath( QStringLiteral( "%1.%2.part" ).arg( fileName, cloudEtag ) );
+
+  mDownloadFileTransfers.insert( fileKey, transfer );
+
+  // Remove old .part files with different etag
+  QDir dir( projectDir );
+  QStringList partFiles = dir.entryList( QStringList() << QStringLiteral( "%1.*.part" ).arg( fileName ), QDir::Files );
+  for ( const QString &partFile : partFiles )
+  {
+    if ( partFile.startsWith( fileName + "." ) && !partFile.contains( cloudEtag ) )
```

Comment:
```suggestion
    if ( partFile.startsWith( fileName + "." ) && !partFile.endsWith( QStringLiteral( "%1.part" ).arg( cloudEtag ) ) )
```

I think this is a bit safer.